### PR TITLE
v3: Update Version Numbers in Docs

### PR DESCRIPTION
**Pull request by nickajacks1**
## Description

Update Go and Fiber version numbers in docs for v3.

## Changes Introduced

List the new features or adjustments introduced in this pull request. Provide details on benchmarks, documentation updates, changelog entries, and if applicable, the migration guide.

- [ ] Benchmarks: Describe any performance benchmarks and improvements related to the changes.
- [x] Documentation Update: Detail the updates made to the documentation and links to the changed files.
- [ ] Changelog/What's New: Include a summary of the additions for the upcoming release notes.
- [ ] Migration Guide: If necessary, provide a guide or steps for users to migrate their existing code to accommodate these changes.
- [ ] API Alignment with Express: Explain how the changes align with the Express API.
- [ ] API Longevity: Discuss the steps taken to ensure that the new or updated APIs are consistent and not prone to breaking changes.
- [ ] Examples: Provide examples demonstrating the new features or changes in action.

## Type of Change

Please delete options that are not relevant.

- [x] Documentation update (changes to documentation)

## Checklist

Before you submit your pull request, please make sure you meet these requirements:

- [ ] Followed the inspiration of the Express.js framework for new functionalities, making them similar in usage.
- [ ] Conducted a self-review of the code and provided comments for complex or critical parts.
- [ ] Updated the documentation in the `/docs/` directory for [Fiber's documentation](https://docs.gofiber.io/).
- [ ] Added or updated unit tests to validate the effectiveness of the changes or new features.
- [ ] Ensured that new and existing unit tests pass locally with the changes.
- [ ] Verified that any new dependencies are essential and have been agreed upon by the maintainers/community.
- [ ] Aimed for optimal performance with minimal allocations in the new code.
- [ ] Provided benchmarks for the new code to analyze and improve upon.

### CODE REVIEW

**Review by sixcolors — APPROVED** [review]

**Comment by gaby** [comment]
@gofiber/maintainers `go1.20` is `EOL` since last week. Do we keep it or bump the min version to `go1.21` ?

https://endoflife.date/go

**Review by gaby — APPROVED** [review]

**Review by efectn — APPROVED** [review]

**Comment by efectn** [comment]
> I think we can keep 1.20 for now, fasthttp is also moving to 1.20 as their min version. A lot of systems don't support 1.21/1.22 yet

I think we should just support 2 latest major versions of Go in v3 but let's discuss it later it should not block the PR

**Pull request merged**
